omod/src/main/java/org/openmrs/module/referenceapplication/filter/RequireLoginLocationFilter.java

```diff
@@ -0,0 +1,121 @@
+/**
+ * The contents of this file are subject to the OpenMRS Public License
+ * Version 1.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ * http://license.openmrs.org
+ *
+ * Software distributed under the License is distributed on an "AS IS"
+ * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the
+ * License for the specific language governing rights and limitations
+ * under the License.
+ *
+ * Copyright (C) OpenMRS, LLC.  All Rights Reserved.
+ */
+package org.openmrs.module.referenceapplication.filter;
+
+import org.apache.commons.lang3.ArrayUtils;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.referenceapplication.ReferenceApplicationConstants;
+import org.openmrs.ui.framework.UiUtils;
+import org.openmrs.ui.framework.WebConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+/**
+ * Intercepts authenticated requests to check if the logged in user has selected a location and will
+ * redirect them to the login page so they can select one in case they haven't yet.
+ */
+public class RequireLoginLocationFilter implements Filter {
```

Comment:
Could you remind me of why this would not work as module filter? Something like this: https://github.com/openmrs/openmrs-module-xforms/blob/master/omod/src/main/resources/config.xml#L500-L507

Reply:
That was a bad hacky way we used to do it historically and it is wrong, those are are 'fake' filters and servlets, servlet api version 3 introduced a new pluggable way to register filters and servlets and it is what we should recommend developers to use.

Reply:
Does the new approach work when any module is installed or restarted?

Reply:
Yes, we used it in legacyui when we are moving it out of core and it has always worked.

Reply:
No it does not work once a module is restarted, stopped, or installed. You will always get the error `Unable to configure mapping for servlet because this servlet context has already been initialized.`

Reply:
That's legacyui module